File: 0206-reverse-linked-list/0206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* previous = nullptr;
        ListNode* current = head;
        if(current == nullptr) return nullptr;
        ListNode* next = current->next;
        while (current->next!=nullptr) {
            current->next = previous;
            previous=current;
            current = next;
            next = next->next;
        }
    current->next = previous;
    return current;
    }
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        node = head
        new_node = head
        nums = []
        while node:
            nums.append(node.val)
            node = node.next
        for num in nums[::-1]:
            new_node.val = num
            new_node = new_node.next
        return head
